beetsplug/hook.py

```diff
@@ -0,0 +1,107 @@
+# This file is part of beets.
+# Copyright 2015, Adrian Sampson.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""Allows custom commands to be run when an event is emitted by beets"""
+from __future__ import (division, absolute_import, print_function,
+                        unicode_literals)
+
+import subprocess
+import sys
+
+from beets.plugins import BeetsPlugin
+
+
+def create_hook_function(log, event, command, shell, substitute_args):
+
+    # TODO: Find a better way of piping STDOUT/STDERR/STDIN between the process
+    #       and the user.
+    #
+    #       The issue with our current method is that we can only pesudo-pipe
+    #       one (two if we count STDERR being piped to STDOUT) stream at a
+    #       time, meaning we can't have both output and input simultaneously.
+    #       This is due to how Popen.std(out/err) works, as
+    #       Popen.std(out/err).readline() waits until a newline has been output
+    #       to the stream before returning.
```

Comment:
Looks like removing the `stdout` and `stderr` arguments does work, I can't believe I didn't try that earlier. Still can't read from `stdin` though, which is a pain but I don't think is that big of an issue (interactive hooks would be better as their own plugins anyway).